### PR TITLE
added *.m for Matlab files

### PR DESCRIPTION
**Pull request by GavriYashar**
Changes proposed in this pull request:
 - basic support for Matlab `*.m` files
 
#3956 partially implemented

### CODE REVIEW

**Comment by RussKie** [comment]
Please create a new issue describing the problem or feature. And in the PR please reference the original issue number.
Thank you

File: Plugins/Statistics/GitStatistics/CodeFile.cs [hunk 1]
```diff
@@ -105,6 +105,8 @@ private void IncrementLineCountsFromLine(string line)
                 NumberCommentsLines++;
             else if (File.Extension.Equals(".cshtml", StringComparison.OrdinalIgnoreCase) && (line.Contains("@*") && line.Contains("*@")))
                 NumberCommentsLines++;
+            else if (File.Extension.Equals(".m", StringComparison.OrdinalIgnoreCase) && line.StartsWith("%"))
```
Comment (RussKie) [inline Plugins/Statistics/GitStatistics/CodeFile.cs]: I haven't used matlab in many years so i'm a bit rusty with its syntax
would `<whitespace> % text` be considered as a comment? if so, it is probably worth while trimming the line before the check

Comment (GavriYashar) [reply Plugins/Statistics/GitStatistics/CodeFile.cs]: yes. Your example would be a valid comment

Comment (GavriYashar) [reply Plugins/Statistics/GitStatistics/CodeFile.cs]: `IncrementLineCountsFromLine` is called in `CountLines` with `line.TrimStart()` so there is no need trim again

**Comment by RussKie** [comment]
@GavriYashar are you planning to address the whole of #3956 in here, or do you wish for this to be merged in as is?

**Comment by GavriYashar** [comment]
@RussKie for now just as is. i'll address these issues another time

**Pull request merged**